File: app/src/main/java/com/pirateman/exercisemanager/ExerciseCriteriaActivity.java
package com.pirateman.exercisemanager;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayout;
import android.view.View;
import android.widget.AutoCompleteTextView;
import android.widget.Button;
import android.widget.TextView;

public class ExerciseCriteriaActivity extends AppCompatActivity {

    //GridLayout layout;
    AutoCompleteTextView tvIncline;
    AutoCompleteTextView tvSpeed;
    AutoCompleteTextView tvDuration;
    AutoCompleteTextView tvIntervals;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_exercise_criteria);

        //ViewDataBinding viewDataBinding = DataBindingUtil.inflate(getLayoutInflater(), R.layout.activity_grid_row);

        //layout = findViewById(R.id.gridLayout);

        Button btn = findViewById(R.id.btnAddInterval);

        tvDuration = findViewById(R.id.actvDuration);
        tvIncline = findViewById(R.id.actvIncline);
        tvSpeed = findViewById(R.id.actvSpeed);
        tvIntervals = findViewById(R.id.actvIntervals);

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String dur = tvDuration.getText().toString();
                String incl = tvIncline.getText().toString();
                String inte = tvIntervals.getText().toString();
                String spe = tvSpeed.getText().toString();

                Intent intent = new Intent(ExerciseCriteriaActivity.this, ExercisePlannerActivity.class);
                intent.putExtra("duration", dur);
                intent.putExtra("incline", incl);
                intent.putExtra("intervals", inte);
                intent.putExtra("speed", spe);

                startActivity(intent);

            }
        });

    }


    public void addRow()
    {
        AutoCompleteTextView textViewActivity = new AutoCompleteTextView(this);
        textViewActivity.setHint("Activity");
        textViewActivity.setTextSize(18f);
        AutoCompleteTextView textViewPace = new AutoCompleteTextView(this);
        textViewPace.setHint("Pace");
        textViewPace.setTextSize(18f);
        AutoCompleteTextView textViewIncline = new AutoCompleteTextView(this);
        textViewIncline.setHint("Incline");
        textViewIncline.setTextSize(18f);
        AutoCompleteTextView textViewDuration = new AutoCompleteTextView(this);
        textViewDuration.setHint("Duration");
        textViewDuration.setTextSize(18f);

//        layout.addView(textViewActivity, setGrid(2, 0));
//        layout.addView(textViewPace, setGrid(2, 1));
//        layout.addView(textViewIncline, setGrid(2, 2));
//        layout.addView(textViewDuration, setGrid(2, 3));

    }

    public GridLayout.LayoutParams setGrid(int row, int col)
    {
        GridLayout.Spec rowSpan = GridLayout.spec(row, 1);
        GridLayout.Spec colspan = GridLayout.spec(col, 1);

        GridLayout.LayoutParams gridParam = new GridLayout.LayoutParams(rowSpan, colspan);

        return gridParam;
    }
}

File: app/src/main/java/com/pirateman/exercisemanager/ExerciseCriteriaActivity.kt
package com.pirateman.exercisemanager

import android.content.Intent
import android.os.Bundle
import android.widget.AutoCompleteTextView
import androidx.appcompat.app.AppCompatActivity
import androidx.gridlayout.widget.GridLayout
import androidx.viewbinding.ViewBinding
import com.pirateman.exercisemanager.databinding.ActivityExerciseCriteriaBinding
import org.koin.android.ext.android.inject


class ExerciseCriteriaActivity : AppCompatActivity() {

    private val binding by inject<ActivityExerciseCriteriaBinding>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val view = binding.root
        setContentView(view)

        binding.btnAddInterval.setOnClickListener {
            val dur = binding.tvDuration.text.toString()
            val incl = binding.tvIncline.text.toString()
            val inte = binding.tvIntervals.text.toString()
            val spe = binding.tvSpeed.text.toString()
            //TODO: use jetpack navigation
            val intent = Intent(this@ExerciseCriteriaActivity, ExercisePlannerActivity::class.java)
            intent.putExtra("duration", dur)
            intent.putExtra("incline", incl)
            intent.putExtra("intervals", inte)
            intent.putExtra("speed", spe)
            startActivity(intent)
        }
    }

    fun addRow() {
        val textViewActivity = AutoCompleteTextView(this)
        textViewActivity.hint = "Activity"
        textViewActivity.textSize = 18f
        val textViewPace = AutoCompleteTextView(this)
        textViewPace.hint = "Pace"
        textViewPace.textSize = 18f
        val textViewIncline = AutoCompleteTextView(this)
        textViewIncline.hint = "Incline"
        textViewIncline.textSize = 18f
        val textViewDuration = AutoCompleteTextView(this)
        textViewDuration.hint = "Duration"
        textViewDuration.textSize = 18f

//        layout.addView(textViewActivity, setGrid(2, 0));
//        layout.addView(textViewPace, setGrid(2, 1));
//        layout.addView(textViewIncline, setGrid(2, 2));
//        layout.addView(textViewDuration, setGrid(2, 3));
    }

    fun setGrid(row: Int, col: Int): GridLayout.LayoutParams {
        val rowSpan = GridLayout.spec(row, 1)
        val colspan = GridLayout.spec(col, 1)
        return GridLayout.LayoutParams(rowSpan, colspan)
    }
}